### ISSUE TITLE
Would it make sense to have different batches with differently sized images?

### ISSUE THREAD

**Issue by vedantroy** [lucidrains/imagen-pytorch#105]
Hi! Since UNets are resolution independent (I think), would it make sense to have batches with differently sized images?
E.g, the first batch might only have 256x256 images. Another batch might only have 512x512 images, and so on.

The benefit would be reducing the amount of data-augmentation that is needed in order to wrangle the images into a consistent set of dimensions, which might increase the data quality? (not sure)

As an aside, I was wondering what people are doing these days for normalizing the data before training.
I'm assuming large datasets do not have images in a consistent 256x256 format, so people must do something to normalize the images into a consistent format.

Is the recommended approach just to center crop everything into a consistent size?

**Comment by Nodja** [comment]
The paper trains the last unet on patches of 64x64 of the original image to 256x256, so there's some merit to what you're saying, but I don't think simply resizing would wield much benefit, it's probably worth attempting tho.

Section 2.5 of the paper
> Our 256 × 256 → 1024 × 1024 super-resolution model trains on 64 × 64 → 256 × 256
crops of the 1024 × 1024 image. To facilitate this, we remove the self-attention layers, however
we keep the text cross-attention layers which we found to be critical. Note that we use text cross attention for both our super-resolution models

**Comment by vedantroy** [comment]
Hmm, so that's for the super-resolution model, but not for the text => image model.
I wonder what they mean by training 64x64 => 256x256 crops.
I assume super-resolution is more like, well increasing the resolution, whereas training to go from a smaller to a larger crop seems more like expanding the boundaries of an image.

**Comment by Nodja** [comment]
> I wonder what they mean by training 64x64 => 256x256 crops.

They cropped a random patch of 256x256 from the original 1024x1024 image and downscaled it to 64x64 and used that as the low res conditioning image with the 256x256 as the target.
